Rendering/WDE.MapSpawns/Rendering/SpawnContextMenu.cs

```diff
@@ -1,23 +1,111 @@
+using System;
 using System.Windows.Input;
 using Prism.Commands;
 using WDE.Common.Services;
+using WDE.Common.Services.MessageBox;
 using WDE.MapSpawns.Models;
 using WDE.MapSpawns.ViewModels;
+using WDE.SqlQueryGenerator;
+using WDE.DatabaseEditors.Services;
+using WDE.DatabaseEditors.Data.Interfaces;
 
 namespace WDE.MapSpawns.Rendering;
 
 public class SpawnContextMenu
 {
     private readonly ISpawnSelectionService spawnSelectionService;
-    
+    private readonly IDatabaseQueryExecutor mySqlExecutor;
+    private readonly ITableDefinitionProvider definitionProvider;
+    private readonly IMessageBoxService messageBoxService;
+
     private ICommand CopyGuidCommand { get; }
+    private ICommand CopyPositionXCommand { get; }
+    private ICommand CopyPositionYCommand { get; }
+    private ICommand CopyPositionZCommand { get; }
+    private ICommand CopyOrientationCommand { get; }
+    private ICommand UpdateValuesCommand { get; }
 
-    public SpawnContextMenu(ISpawnSelectionService spawnSelectionService,
-        IClipboardService clipboardService)
+    public SpawnContextMenu(
+        ISpawnSelectionService spawnSelectionService,
+        IClipboardService clipboardService,
+        ITableEditorPickerService tableEditorPickerService,
+        IDatabaseQueryExecutor mySqlExecutor,
+        ITableDefinitionProvider definitionProvider,
+        IMessageBoxService messageBoxService)
     {
         this.spawnSelectionService = spawnSelectionService;
+        this.mySqlExecutor = mySqlExecutor;
+        this.definitionProvider = definitionProvider;
+        this.messageBoxService = messageBoxService;
 
         CopyGuidCommand = new DelegateCommand<SpawnInstance>(inst => clipboardService.SetText(inst.Guid.ToString()));
+        CopyPositionXCommand = new DelegateCommand<SpawnInstance>(inst => clipboardService.SetText(inst.WorldObject!.Position.X.ToString()));
```

Comment:
I wonder, is it worth to have this as three separate menu items instead of one "Copy Position" which would copy comma separated values? `x, y, z` Just wondering, if it is better to have them separate I am fine I guess.

Reply:
Before implementing the "Update values" option i just had a copy option for all the values that needed updating, so I could just click "Copy foo", open DB Table Editor, and paste the copied value into the column foo. I deleted most of the options as those got obsolete because of the new "Update values", but figured the positions could still be useful to keep. I think you are right, combining them, now that there is the new save option, is probably the way to go.